float32 range
float32 angle
float32 xrel
float32 yrel
bool segment_status
